src/intrinsics/prepack/global.js

```diff
@@ -131,13 +119,14 @@ export default function(realm: Realm): void {
   global.$DefineOwnProperty("__optimize", {
     value: new NativeFunctionValue(realm, "global.__optimize", "__optimize", 1, (context, [value, argModelString]) => {
       // only optimize functions for now
+      let argModel;
       if (argModelString !== undefined) {
         let argModelError;
         if (argModelString instanceof StringValue) {
           try {
             // result here is ignored as the main point here is to
             // check and produce error
-            JSON.parse(argModelString.value);
+            argModel = (JSON.parse(argModelString.value): ArgModel);
```

Comment:
Not your fault, but this statement seems unsafe. A TODO comment and an issue would be great.